### PR TITLE
Fix(node-fetch): wd 11834 fix node fetch vulnerability

### PR DESCRIPTION
**Pull request by Worvast**
Regular Expression Denial of Service (ReDoS) Affecting [node-fetch Open this link in a new tab ](https://snyk.io/vuln/npm%3Anode-fetch)package, versions <3.2.10

    Level: high

    Exploit Maturity: Proof of concept

    Attack Complexity: Low

    Availability: High

     

Introduced: 31 Jul 2022
How to fix?

Upgrade node-fetch to version 3.2.10 or higher.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1,10487 +1,8 @@
 {
   "name": "@devoinc/browser-sdk",
-  "version": "3.0.3",
-  "lockfileVersion": 2,
+  "version": "3.0.4",
+  "lockfileVersion": 1,
```
Comment (lbermudez) [inline package-lock.json:4]: Is this correct @Worvast ?

Comment (Worvast) [reply package-lock.json:4]: yes we need it to pass test in v12

Comment (tripu) [reply package-lock.json:4]: :-1: to downgrade the lockfile version.

I doubt you need that. Make sure you do `npm i -g npm@latest` or `npm i -g npm@7` as part of (at the beginning of) your builds, and you'll have `lockfileVersion` 2 even when using Node.js 12.

File: package.json [hunk 2]
```diff
@@ -72,6 +71,7 @@
     "mocha": "^9.2.2",
     "mocha-each": "^2.0.1",
     "msw": "^0.39.2",
+    "node-fetch": "^3.2.10",
```
Comment (tripu) [inline package.json:74]: You have moved `node-fetch` from dependencies to devDependencies. It won't be installed in production or when packaging (only for development). Is that intended?

**Review by tripu — CHANGES_REQUESTED** [review]

File: package-lock.json (hunk 1 above)
Comment (Worvast) [reply package-lock.json]: when i create directly for v2 fail in the tests, here and in local, if i update it from v1 works, look like its ok now i hope works the actions

File: package.json (hunk 2 above)
Comment (Worvast) [reply package.json:74]: Yes. Its only used for tests.

**Review by tripu — APPROVED** [review]

File: package-lock.json (hunk 1 above)
Comment (lbermudez) [reply package-lock.json]: But, the package.json has:
```
"engines": {
    "node": ">=14",
    "npm": ">=7"
  }
```

Comment (tripu) [reply package-lock.json]: Yes, @lbermudez. @Worvast changed it back: 31cd224e134f5c99fba09fe544e009d62b77273d

**Comment by tripu** [comment]
(Just deleted the branch)

**Pull request merged**